File: features/lead/src/main/java/com/nivasafinance/features/lead/service/LeadWriteService.java
package com.nivasafinance.features.lead.service;

import com.nivasafinance.features.lead.dto.*;

import java.util.UUID;

public interface LeadWriteService {
    CreateLeadResponse createLead(CreateLeadRequest request);
    void updateLead(UUID leadIdentifier, UpdateLeadRequest request);
    // to update updatedby and updated at
    void touchLead(UUID leadIdentifier);
    void updatePreliminaryDetails(UUID leadIdentifier, UpdatePreliminaryDetailsRequest request);
    void updateCreditDetails(UUID leadIdentifier, UpdateCreditDetailsRequest request);
    void updateProposedDetails(UUID leadIdentifier, UpdateProposedDetailsRequest request);
    void updatePropertyDetails(UUID leadIdentifier, UpdatePropertyDetailsRequest request);
    void updateSourcingDetails(UUID leadIdentifier, UpdateSourcingDetailsRequest request);
    void updateCallDetails(UUID leadIdentifier, UpdateCallDetailsRequest request);
    void updateDisbursementDetails(UUID leadIdentifier, UpdateDisbursementDetailsRequest request);
    void createTranche(UUID leadIdentifier, CreateTrancheRequest request);
    void updateTranche(UUID leadIdentifier, UUID trancheIdentifier, UpdateTrancheRequest request);
    void deleteTranche(UUID leadIdentifier, UUID trancheIdentifier);
    void rejectLead(UUID leadIdentifier, RejectLeadRequest request);
    void undoRejectLead(UUID leadIdentifier);
    void withdrawLead(UUID leadIdentifier, WithdrawLeadRequest request);
    void completeLead(UUID leadIdentifier);
    void onholdLead(UUID leadIdentifier, OnholdLeadRequest request);
    void resumeLead(UUID leadIdentifier);
    void dropoffLead(UUID leadIdentifier, DropoffLeadRequest request);
    void patchPropertyDetails(UUID leadIdentifier, PatchPropertyDetailsRequest request);
    void patchIncomeObligationDetails(UUID leadIdentifier, PatchIncomeAndObligationRequest request);
    void patchLead(UUID leadIdentifier, PatchLeadRequest request);
    void deleteLead(UUID leadIdentifier);
    void undoDeleteLead(UUID leadIdentifier);
}

File: features/lead/src/main/java/com/nivasafinance/features/lead/lead/service/LeadWriteService.kt
package com.nivasafinance.features.lead.lead.service

import com.nivasafinance.features.lead.lead.dto.LeadCreateRequest
import com.nivasafinance.features.lead.lead.dto.LeadPatchRequest
import com.nivasafinance.features.lead.lead.dto.LeadResponse
import data.Identifier
import java.util.UUID

interface LeadWriteService {
    fun createLead(request: LeadCreateRequest): LeadResponse
    fun patchLead(id: UUID, request: LeadPatchRequest)
    fun saveIdentifier(id: UUID, applicantId: UUID, addIdentifier: Identifier)
    fun updateIdentifier(id: UUID, applicantId: UUID, identifierId: UUID, addIdentifier: Identifier)
}
